src/Barricade.h

```diff
@@ -0,0 +1,42 @@
+#ifndef BARRICADE_H
+#define BARRICADE_H
+#include <string>
+#include "LTexture.h"
+#include "HitBox.h"
+#include "Object.h"
+#include "CollisionHandler.h"
+#include "Inventory.h"
+#include <math.h>
+#include <vector>
+#include <SDL2/SDL.h>
+#include "Window.h"
+
+class Barricade : public Object {
+public:
+
+	void move(float, float, float, float, CollisionHandler*); // Moves Zombie
+
+	void onCollision();
+
+	void collidingProjectile(int damage);
+
+	void setAngle(double a);//sets angle of sprite to
+
+	double getAngle(); //returns sprites angle
+
+	bool isPlaceable();
+
+	bool checkPlaceablePosition(float, float, float, float, CollisionHandler*);
+
+	Barricade();
+	virtual ~Barricade();
+
+private:
```

Comment:
Defaults should be specified in the constructor not in the definitions of members. When implementing the constructor the with defaults it should be also using member initializers